robot_calibration/src/ground_plane_finder.cpp

```diff
@@ -131,16 +131,19 @@ bool GroundPlaneFinder::find(robot_calibration_msgs::CalibrationData * msg)
   sensor_msgs::PointCloud2Iterator<float> iter_cloud(cloud_, "x");
 
   // Set msg size
-  msg->observations.resize(2);
-  msg->observations[0].sensor_name = camera_sensor_name_;
-  msg->observations[0].features.resize(points_total);
-  msg->observations[1].sensor_name = chain_sensor_name_;
-  msg->observations[1].features.resize(points_total);
-
-  size_t step = cloud_.width/points_total;
+  int idx_cam = msg->observations.size() + 0;
+  int idx_chain = msg->observations.size() + 1;
+  msg->observations.resize(msg->observations.size() + 2);
+  msg->observations[idx_cam].sensor_name = camera_sensor_name_;
+  msg->observations[idx_chain].sensor_name = chain_sensor_name_;
+  
+  msg->observations[idx_cam].features.resize(points_total);
+  msg->observations[idx_chain].features.resize(points_total);
+
+  size_t step = cloud_.width/(1.5 * points_total);
   size_t k = 0;
 
-  for (size_t i = step; i < cloud_.width; i +=step)
+  for (size_t i = step; i < cloud_.width/1.5; i +=step)
```

Comment:
what is up with this 1.5? this seems like a change in performance, we should probably put it in a separate commit